test/powershell/engine/Module/TestModuleManifest.ps1

```diff
@@ -24,32 +24,74 @@ Describe "Test-ModuleManifest tests" -tags "CI" {
         Test-ModuleManifest -Path $testModulePath -ErrorAction Stop | Should BeOfType System.Management.Automation.PSModuleInfo
     }
 
-    It "module manifest containing missing files returns error" {
+    It "module manifest containing missing files returns error" -TestCases `
+        @{parameter = "RequiredAssemblies"; error = "Modules_InvalidRequiredAssembliesInModuleManifest"},
+        @{parameter = "NestedModules"; error = "Modules_InvalidNestedModuleinModuleManifest"},
+        @{parameter = "RequiredModules"; error = "Modules_InvalidRequiredModulesinModuleManifest"},
+        @{parameter = "FileList"; error = "Modules_InvalidFilePathinModuleManifest"},
+        @{parameter = "ModuleList"; error = "Modules_InvalidModuleListinModuleManifest"},
+        @{parameter = "TypesToProcess"; error = "Modules_InvalidManifest"},
+        @{parameter = "FormatsToProcess"; error = "Modules_InvalidManifest"},
+        @{parameter = "RootModule"; error = "Modules_InvalidRootModuleInModuleManifest"},
+        @{parameter = "ScriptsToProcess"; error = "Modules_InvalidManifest"} {
+
+        param ($parameter, $error)
 
         New-Item -ItemType Directory -Path testdrive:/module
         New-Item -ItemType Directory -Path testdrive:/module/foo
-        Set-Content -Value "" -Path testdrive:/module/foo/bar.psm1
+        New-Item -ItemType File -Path testdrive:/module/foo/bar.psm1
         $testModulePath = "testdrive:/module/test.psd1"
 
-        $parametersAndErrors = @{"RequiredAssemblies"="Modules_InvalidRequiredAssembliesInModuleManifest";
-            "NestedModules"="Modules_InvalidNestedModuleinModuleManifest";
-            "RequiredModules"="Modules_InvalidRequiredModulesinModuleManifest";
-            "FileList"="Modules_InvalidFilePathinModuleManifest";
-            "ModuleList"="Modules_InvalidModuleListinModuleManifest";
-            "TypesToProcess"="Modules_InvalidManifest";
-            "FormatsToProcess"="Modules_InvalidManifest";
-            #"RootModule"="Modules_InvalidManifest";
-            "ScriptsToProcess"="Modules_InvalidManifest"
+        $args = @{$parameter = "doesnotexist.psm1"}
+        New-ModuleManifest -Path $testModulePath @args
+        Test-Path $testModulePath | Should Be $true
+        [string]$errorId = "$error,Microsoft.PowerShell.Commands.TestModuleManifestCommand"
+
+        { Test-ModuleManifest -Path $testModulePath -ErrorAction Stop } | ShouldBeErrorId $errorId
+    }
+
+    It "module manifest containing valid rootmodule succeeds" -TestCases `
+        @{rootModuleValue = $null},
+        @{rootModuleValue = ""},
+        @{rootModuleValue = "foo.psm1"},
+        @{rootModuleValue = "foo.dll"} {
+
+        param($rootModuleValue)
+
+        New-Item -ItemType Directory -Path testdrive:/module
+        $testModulePath = "testdrive:/module/test.psd1"
+
+        if ($rootModuleValue -ne $null -and $rootModuleValue -ne "")
```

Comment:
It seems it is better to us to split the tests into two and remove the conditions.